Travis config improvements

CvX opened the pull request:
* Unified indentation
* Merged duplicate keys
* Extracted Ember tests to a separate job (just trying it out, we'll see if it works at all)

jtgeibel commented:
The cleanups in the first 2 commits look really good, thanks!  I'm also unsure about the final commit, as the clippy run usually takes the most time and there is a lot of overhead (like `cargo install diesel_cli ...`) to running another build.

Turbo87 commented:
I tend to agree. Since we still run `cargo install` even for the Ember job this doesn't look like a win to me 🤔 

The other two commits look good though, if you can remove the third one from the PR I guess we can merge.

CvX commented:
I've removed the failed speedup experiment and added node_modules caching.

It seems that travis didn't pick up on force-pushed changes since there are still 5 build jobs instead of 4 so I'm closing this PR and creating a new one.
